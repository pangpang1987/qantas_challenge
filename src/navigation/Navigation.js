import React, { Fragment } from "react";
import PropTypes from "prop-types";

import { NavigationBar, Content, Link } from "./styled";

class Navigation extends React.Component {
  render() {
    const { children } = this.props;

    return (
      <Fragment>
        <NavigationBar>
          <Link exact to="/" activeClassName="isActive">
            Home
          </Link>
          <Link exact to="/faqs" activeClassName="isActive">
            Faqs
          </Link>
        </NavigationBar>
        <Content>{children}</Content>
      </Fragment>
    );
  }
}

Navigation.propTypes = {
  children: PropTypes.node.isRequired
};

export default Navigation;
