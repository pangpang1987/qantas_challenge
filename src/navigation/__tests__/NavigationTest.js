import React from "react";
import Navigation from "../Navigation";
// import renderer from "react-test-renderer";

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

it("renders initial state correctly", () => {
  const NavigationComponent = shallow(<Navigation>test</Navigation>);
  const tree = toJson(NavigationComponent);
  expect(tree).toMatchSnapshot();
});
