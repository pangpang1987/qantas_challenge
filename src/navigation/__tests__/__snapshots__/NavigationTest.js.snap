// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders initial state correctly 1`] = `
<React.Fragment>
  <styled.div>
    <Styled(NavLink)
      activeClassName="isActive"
      exact={true}
      to="/"
    >
      Home
    </Styled(NavLink)>
    <Styled(NavLink)
      activeClassName="isActive"
      exact={true}
      to="/faqs"
    >
      Faqs
    </Styled(NavLink)>
  </styled.div>
  <styled.div>
    test
  </styled.div>
</React.Fragment>
`;
